packages/angular/ssr/src/routes/ng-routes.ts

```diff
@@ -124,19 +127,23 @@ async function* traverseRoutesConfig(options: {
         if (!matchedMetaData) {
           yield {
             error:
-              `The '${currentRoutePath}' route does not match any route defined in the server routing configuration. ` +
+              `The '${stripLeadingSlash(currentRoutePath)}' route does not match any route defined in the server routing configuration. ` +
               'Please ensure this route is added to the server routing configuration.',
           };
 
           continue;
         }
+
+        matchedMetaData.matched = true;
       }
 
       const metadata: ServerConfigRouteTreeNodeMetadata = {
         ...matchedMetaData,
         route: currentRoutePath,
       };
 
+      delete metadata.matched;
```

Comment:
The `delete` operation used to cause deopt in VM, may be use `null` or `undefined` instead?
```suggestion
      metadata.matched = null;
```

Reply:
This is done during build time so I am less worried about that, if we set undefined or null, we’d need to filter it out so not to include it in the manifest